roles/repository/tasks/main.yml

```
@@ -41,6 +41,15 @@
           JAVA_TOOL_OPTIONS: "{{ temp_array }}"
       when: temp_array|length > 0
 
+- name: Install curl except Redhat 9
+  become: true
+  ansible.builtin.package:
+    name: curl
+    state: present
+  when:
+    - ansible_os_family == 'RedHat'
+    - ansible_distribution_major_version is not version('9', '>=')
```

```suggestion
    - ansible_distribution_major_version is version('9', '<')
```

to make De Morgan happy